tcases-lib/src/main/java/org/cornutum/tcases/generator/io/GeneratorSetDocWriter.java

```diff
@@ -100,107 +98,89 @@ protected void writeGenerator( String function, ITestCaseGenerator generator)
    */
   protected void writeTupleGenerator( String function, TupleGenerator generator)
     {
-    writer_.writeTagStart( TUPLEGENERATOR_TAG);
-    writer_.writeAttribute( FUNCTION_ATR, function);
-
     Long seed = generator.getRandomSeed();
-    if( seed != null)
-      {
-      writer_.writeAttribute( SEED_ATR, String.valueOf( seed));      
-      }
-
-    writer_.writeAttribute( TUPLES_ATR, String.valueOf( generator.getDefaultTupleSize()));      
-    writer_.writeTagEnd();
-
-    writer_.indent();
-    for( TupleCombiner combiner : generator.getCombiners())
-      {
-      writeCombiner( combiner);
-      }
-    writer_.unindent();
     
-    writer_.writeElementEnd( TUPLEGENERATOR_TAG);
+    writer_.writeElement(
+      TUPLEGENERATOR_TAG,
+
+      MapBuilder
```

Comment:
actually, if you like the builder Pattern for this and consistently use it, you might as well have a Builder for the whole XML tag.

Reply:
Even better! Why didn't I think of this?